File: Service-Hub/ServiceHub.BL/DTOs/RegistrationDTO.cs

using ServiceHub.DAL.Enums;
using System.ComponentModel.DataAnnotations;

namespace ServiceHub.BL.DTOs
{
    public class RegistrationDTO
    {
        [Required]
        public string? FullName { get; set; }
        [Required,EmailAddress]
        public string? Email { get; set; }
        [Required]
        public string Password { get; set; }
        [Required]
        [Compare("Password")]
        public string? ConfirmPassword { get; set; }
        public Role Role { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.Now;
        public int DistrictId { get; set; }



    }
}

File: Service-Hub-Frontend/src/app/Models/registration-dto.ts
import { Role } from "../enums/role";

export class RegisterationDTO
{
  constructor(
    public fullName: string,
    public email: string,
    public password:string,
    public confirmPassword:string,
    public role:Role,
    public districtId:number ,
    public jobId:number
    ) { }
}
